### PR TITLE
[CDEC-400] Reunite orphan Bi instances in update

### PR DESCRIPTION
**Pull request by intricate**
## Description

The directory `update/src/Pos/Binary/Update` has three files containing orphan Bi instances. This PR reunites those instances with their types.

Note: In `Pos.Binary.Update.Relay`, there were some orphan Bi instances for types which are defined in `infra` (`Pos.Infra.Communication.Types.Relay`) so I moved them over to the appropriate module.

## Linked issue

[https://iohk.myjetbrains.com/youtrack/issue/CDEC-400](https://iohk.myjetbrains.com/youtrack/issue/CDEC-400)
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: update/src/Pos/Update/Poll/Types.hs [hunk 1]
```diff
@@ -119,6 +130,17 @@ type LocalVotes = HashMap UpId (HashMap PublicKey UpdateVote)
...
 ----------------------------------------------------------------------------
 
+-- | Extra data required by wallet, stored in UndecidedProposalState
+data UpsExtra = UpsExtra
+    { ueProposedBlk :: !HeaderHash
+    -- ^ Block in which this update was proposed
+    } deriving (Show, Generic, Eq)
+
+deriveSimpleBi ''UpsExtra [
+  Cons 'UpsExtra [
```
Comment (erikd) [inline update/src/Pos/Update/Poll/Types.hs]: We should try to stick to the 4 space indent standard.

File: update/src/Pos/Update/Poll/Types.hs [hunk 2]
```diff
@@ -136,12 +158,30 @@ data UndecidedProposalState = UndecidedProposalState
       -- ^ Extra data
     } deriving (Show, Generic, Eq)
 
--- | Extra data required by wallet, stored in UndecidedProposalState
-data UpsExtra = UpsExtra
-    { ueProposedBlk :: !HeaderHash
-    -- ^ Block in which this update was proposed
+deriveSimpleBi ''UndecidedProposalState [
+  Cons 'UndecidedProposalState [
```
Comment (erikd) [inline update/src/Pos/Update/Poll/Types.hs]: Indentation

File: update/src/Pos/Update/Poll/Types.hs [hunk 3]
```diff
@@ -269,6 +340,15 @@ bvsMaxBlockSize = bvmMaxBlockSize . bvsModifier
 data PrevValue a = PrevValue a | NoExist
     deriving (Generic, Show, Eq)
 
+instance Bi a => Bi (PrevValue a) where
+  encode (PrevValue a) = encodeListLen 1 <> encode a
```
Comment (erikd) [inline update/src/Pos/Update/Poll/Types.hs]: Indentation

File: update/src/Pos/Update/Poll/Types.hs (hunk 1 above)
Comment (intricate) [reply update/src/Pos/Update/Poll/Types.hs]: Not sure what happened here. Will fix these.

**Comment by erikd** [comment]
When you sort out the indentation, please feel free to dismiss my review.

**Pull request merged**